Sources/WhoopDIKit/Container/Container.swift

```diff
@@ -54,28 +60,30 @@ public final class Container {
     public func inject<T>(_ name: String? = nil,
                           params: Any? = nil,
                           _ localDefinition: (DependencyModule) -> Void) -> T {
-        guard localServiceDict == nil else {
+        guard !isLocalInjectActive else {
             fatalError("Nesting WhoopDI.inject with local definitions is not currently supported")
         }
+        
+        isLocalInjectActive = true
+        defer {
+            isLocalInjectActive = false
```

Comment:
Could this not also have a memory safety race? I wonder if we should have this be part of the `ThreadSafeDependencyGraph` and ony check that

Reply:
_maybe_? I think boolean updates are atomic but let me do some research on this.

Reply:
Oh nevermind I see what you are saying.

Reply:
➕ This looks like, we can hit the fatalError above here.

Reply:
I'll move this down into the `aquireDependencyGraph` block which is internally protected by a recursive lock.